Python/Product/Analysis/LanguageServer/Settings.cs

```diff
@@ -15,24 +15,62 @@
 // permissions and limitations under the License.
 
 using System;
+using System.Collections.Generic;
 
 namespace Microsoft.PythonTools.Analysis.LanguageServer {
-    internal class LanguageServerSettings {
-        public AutoCompleteSettings autoComplete;
-        public DiagnosticsSettings diagnostics;
+    public sealed class LanguageServerSettings {
+        public class PythonAnalysisOptions {
+            public bool openFilesOnly;
+            public object[] errors = Array.Empty<object>();
+            public object[] warnings = Array.Empty<object>();
+            public object[] information = Array.Empty<object>();
+            public object[] disabled = Array.Empty<object>();
 
-        [Serializable]
-        internal class AutoCompleteSettings {
-            public bool showAdvancedMembers;
-        }
+            private HashSet<object> _errors;
+            private HashSet<object> _warnings;
+            private HashSet<object> _information;
+            private HashSet<object> _disabled;
```

Comment:
Initialize these in the constructor?

Reply:
Or better yet, remap them to a `Dictionary<code, severity>` so that getting the overridden severity is a simple `TryGetValue`. (But leave the interop schema as the four arrays.)

Reply:
Sure. I just thought they may be empty most of the time so avoided creation of hashsets

Reply:
They can still be initialized to null if they'd be empty. Doing it lazily doesn't really achieve anything here apart from keeping the original arrays alive for longer.